packages/x-charts/src/context/InteractionProvider.tsx

```diff
@@ -1,132 +1,41 @@
 'use client';
 import * as React from 'react';
 import { ChartItemIdentifier, ChartSeriesType } from '../models/seriesType/config';
+import { useCharts } from '../internals/useCharts';
+import { ChartsStore } from '../internals/plugins/utils/ChartsStore';
 
 export interface InteractionProviderProps {
   children: React.ReactNode;
 }
 
 export type ItemInteractionData<T extends ChartSeriesType> = ChartItemIdentifier<T>;
 
-export type AxisInteractionData = {
-  x: null | {
-    value: number | Date | string;
-    // Set to -1 if no index.
-    index: number;
-  };
-  y: null | {
-    value: number | Date | string;
-    // Set to -1 if no index.
-    index: number;
-  };
-};
-
-type InteractionActions<T extends ChartSeriesType = ChartSeriesType> =
-  | {
-      type: 'enterItem';
-      data: ItemInteractionData<T>;
-    }
-  | {
-      type: 'leaveItem';
-      data: Partial<ItemInteractionData<T>>;
-    }
-  | {
-      type: 'exitChart';
-    }
-  | {
-      type: 'updateVoronoiUsage';
-      useVoronoiInteraction: boolean;
-    }
-  | {
-      type: 'updateAxis';
-      data: AxisInteractionData;
-    };
-
-type InteractionState = {
-  /**
-   * The item currently interacting.
-   */
-  item: null | ItemInteractionData<ChartSeriesType>;
-  /**
-   * The x- and y-axes currently interacting.
-   */
-  axis: AxisInteractionData;
-  /**
-   * Set to `true` when `VoronoiHandler` is active.
-   * Used to prevent collision with mouseEnter events.
-   */
-  useVoronoiInteraction: boolean;
-  dispatch: React.Dispatch<InteractionActions>;
-};
-
-export const InteractionContext = React.createContext<InteractionState>({
-  item: null,
-  axis: { x: null, y: null },
-  useVoronoiInteraction: false,
-  dispatch: () => null,
-});
+export const ChartsContext = React.createContext<{ store: ChartsStore } | null>(null);
```

Comment:
Is the goal to have a single store eventually? Or even a main store and fewer secondary stores? 

If that is the case it would be a good idea to rename the file here, and move it to its own folder, as well as having more meaningful names for the context and the `useStore` as well.

Having everything related to these in a single folder would also help with discoverability. Right now we have files in `internals/plugins/models`, `internals/plugins/utils`, `internals/plugins`, `hooks` and `context` that are all kinda related to eachother.